engine/src/main/java/io/camunda/zeebe/process/test/engine/GrpcResponseWriter.java

```diff
@@ -109,6 +114,9 @@ public void tryWriteResponse(final int requestStreamId, final long requestId) {
       final GeneratedMessageV3 response =
           responseMapper.map(request.requestType(), valueBufferView, key, intent);
       sendResponse(request, response);
+      if (requestListener != null) {
+        requestListener.accept(intent);
+      }
```

Comment:
💭 One thing that just came to my mind: what happens if the sendResponse fails? Do we still want to invoke the listener? Should we invoke it earlier? Otherwise, we won't know that it was invoked for the intent just because the method invocation failed. That might might lead to wrong assumptions, e.g. a response for the intent was not invoked although it was and we just didn't notice.